src/components/editNotePage/EditNotePage.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { Component, Props } from 'react';
+// tslint:disable-next-line:no-implicit-dependencies
+import { RouteComponentProps } from 'react-router';
+import { Link } from 'react-router-dom';
+import { debounce } from '../../helpers/debounce';
+import { httpMethod, httpRequest } from '../../helpers/httpRequest';
+import { INote } from '../../interfaces/note';
+import NoteEditing from '../noteEditing/NoteEditing';
+
+type EditNoteProps = Props<{}> & RouteComponentProps<{id: string}>;
+type State = {note: INote};
+
+export default class EditNotePage extends Component<EditNoteProps, State> {
+
+  public debounceEditNote = debounce(200, this.editNote);
+
+  constructor(props: EditNoteProps) {
+    super(props);
+    this.state = {note: {description: '', title: '', created: new Date(), isDone: false, isArchived: false}};
+    this.handleEditNote = this.handleEditNote.bind(this);
+  }
+
+  public componentDidMount() {
+    this.loadNote(this.props.match.params.id);
+  }
+
+  public loadNote(id: string) {
+    httpRequest(httpMethod.get, 'notes/' + id)
+    .then((res) => res.json())
+    .then((note) => this.setState({note}))
+    .catch((er: Error) => alert(`Er ${er}`));
+  }
+
+  public handleEditNote(change: {title?: string, description?: string}) {
+    this.setState({note: {...this.state.note, ...change}});
```

Comment:
Consider adding the following plugin to your tslint:
https://github.com/buzinas/tslint-eslint-rules

Plain tslint is missing many major ESLint rules.

Particularly, in this case it's missing the following rule:
https://eslint.org/docs/rules/object-curly-spacing

which would force you to format objects like so:
```js
this.setState({ note: { ...this.state.note, ...change } });
```
making them more readable

Reply:
added this rule and aligned everything